linera-chain/src/unit_tests/data_types_tests.rs

```diff
@@ -59,7 +59,7 @@ fn test_hashes() {
     let confirmed_hashed = Hashed::new(ConfirmedBlock::new(executed_block.clone()));
     let validated_hashed = Hashed::new(ValidatedBlock::new(executed_block));
 
-    assert_ne!(confirmed_hashed.hash(), validated_hashed.hash());
+    assert_eq!(confirmed_hashed.hash(), validated_hashed.hash());
```

Comment:
Maybe we should have a test for the concern that this wan was originally meant to address: Check that you can't use a signature from a vote/certificate for a validated block inside a vote/certificate for the same _confirmed_ block; signature verification should fail.

Reply:
In the test above we check that votes for Confirmed and Validated blocks (containing the same executed block inside) are not equal. Do you want me to extend that to also check that signatures for those votes aren't interchangeable?

Reply:
Ah, right, I forgot about the other test.
But yes, I think that would be best: Check that the signature for the validated block doesn't work as a signature for the confirmed block.

Reply:
That will pass though b/c when signing we sign `LiteValue` type which contains only the _hash_ of the block (so the `ExecutedBlock` type) and we add the _kind_ of the block/certificate separately in `LiteVote`.

Separately from that - I just noticed that [we don't sign `ChainId` ](https://github.com/linera-io/linera-protocol/blob/7799087241b2bc667244c60782c73cf82fa7bc60/linera-chain/src/data_types.rs#L822-L823) and I think we should.

Reply:
That's what I mean: I'd add that test just to make sure we never accidentally remove the kind from the sign data.

The block already contains chain ID, so that should be fine.

Reply:
Got it now, I thought you meant to change how the types look like.

Right, hash already "contains" the chain_id b/c it was part of the struct. Maybe we can "lift" it to `LiteVote` type so that the "metadata" of where the vote comes from is easily available. RN it's impossible to know what chain that vote is for/from.

Reply:
More generally, you're saying: Maybe it would be more useful if the lite vote contained the (TBD) _header_ of the block rather than just the _hash_?

Definitely worth discussing, but probably something for a separate PR?

Reply:
I meant that we keep the hash (of the block) but add the chainId explicitly, alongside, rest of the data. But maybe you're right and it'll be more useful to include the whole block header there. That's how it'll work when bridging (and validating w/ light clients) so it makes sense to have the same approach everywhere.

Definitely not in this PR.